python/ray/train/base_trainer.py

```diff
@@ -298,13 +307,15 @@ def training_loop(self):
         Returns:
             BaseTrainer: A restored instance of the class that is calling this method.
         """
-        if not cls.can_restore(path):
+        if not cls.can_restore(path, storage_filesystem):
             raise ValueError(
                 f"Invalid restore path: {path}. Make sure that this path exists and "
                 "is the experiment directory that results from a call to "
                 "`trainer.fit()`."
             )
-        trainer_state_path = cls._maybe_sync_down_trainer_state(path)
+        trainer_state_path = cls._maybe_sync_down_trainer_state(
```

Comment:
Instead of syncing down the state, why not change line 321 below to call filesystem.open() to read the data directly from the filesystem?